code/datums/mutations/actions.dm

```diff
@@ -8,6 +8,95 @@
 	power = /obj/effect/proc_holder/spell/targeted/telepathy
 	instability = 10
 
+
+/datum/mutation/human/olfaction
+	name = "Transcendent Olfaction"
+	desc = "Your sense of smell is comparable to that of a canine."
+	quality = POSITIVE
+	difficulty = 12
+	text_gain_indication = "<span class='notice'>Smells begin to make more sense...</span>"
+	text_lose_indication = "<span class='notice'>Your sense of smell goes back to normal.</span>"
+	power = /obj/effect/proc_holder/spell/targeted/olfaction
+	instability = 30
+	var/reek = 200
+
+/datum/mutation/human/olfaction/on_life()
+	var/hygiene_now = owner.hygiene
+
+	if(hygiene_now < 100 && prob(5))
+		owner.adjust_disgust(rand(3,5))
+	if(hygiene_now < HYGIENE_LEVEL_DIRTY && prob(50))
+		to_chat(owner,"<span class='danger'>You get a whiff of your stench and feel sick!</span>")
+		owner.adjust_disgust(rand(5,10))
+
+	if(hygiene_now < HYGIENE_LEVEL_NORMAL && reek >= HYGIENE_LEVEL_NORMAL)
+		to_chat(owner,"<span class='warning'>Your inhumanly strong nose picks up a faint odor. Maybe you should shower soon.</span>")
+	if(hygiene_now < 150 && reek >= 150)
+		to_chat(owner,"<span class='warning'>Your odor is getting bad, what with you having a super-nose and all.</span>")
+	if(hygiene_now < 100 && reek >= 100)
+		to_chat(owner,"<span class='danger'>Your odor begins to make you gag. You silently curse your godly nose. You should really get clean!</span>")
+	if(hygiene_now < HYGIENE_LEVEL_DIRTY && reek >= HYGIENE_LEVEL_DIRTY)
+		to_chat(owner,"<span class='danger'>Your super-nose is fed up with your stench. You could really use a shower.</span>")
```

Comment:
Shouldn't these be else ifs? It seems you'll be getting spammed all 4 at the highest level

Reply:
No. The `reek` var tracks your hygiene amount from the previous tick so that these messages are only shown once when you pass a threshold. If you somehow pass two thresholds in a tick, I'd like multiple messages to show.